modules/EvseManager/EvseManager.cpp

```diff
@@ -118,8 +120,17 @@ void EvseManager::init() {
     if (hlc_enabled) {
         if (config.charge_mode == "DC") {
             // subscribe to run time updates for real initial values (and changes e.g. due to de-rating)
-            r_powersupply_DC[0]->subscribe_capabilities(
-                [this](const auto& caps) { update_powersupply_capabilities(caps); });
+            r_powersupply_DC[0]->subscribe_capabilities([this](const auto& caps) {
+                update_powersupply_capabilities(caps);
+                const bool dc_was_updated = update_supported_energy_transfers(types::iso15118::EnergyTransferMode::DC);
+                bool dc_bpt_was_updated = false;
+                if (caps.bidirectional) {
+                    dc_bpt_was_updated = update_supported_energy_transfers(types::iso15118::EnergyTransferMode::DC_BPT);
+                }
```

Comment:
You can also write like this:
```suggestion
                const bool dc_bpt_was_updated = caps.bidirectional ? update_supported_energy_transfers(types::iso15118::EnergyTransferMode::DC_BPT) : false;
```